webui/src/app/app.component.spec.ts

```diff
@@ -82,4 +84,23 @@ describe('AppComponent', () => {
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should set avatar image url', () => {
+    const expectedUrl = 'http://example.com';
+    const userSpy = {photoURL: expectedUrl};
+
+    expect(onAuthStateChangedSpy).toHaveBeenCalledWith(jasmine.any(Function));
+    const [ cb ] = onAuthStateChangedSpy.calls.mostRecent().args;
+    cb(userSpy);
+
+    expect(component.avatarImageUrl).toBe(expectedUrl);
+  });
+
+  it('should not set avatar image url when user is null', () => {
+    expect(onAuthStateChangedSpy).toHaveBeenCalledWith(jasmine.any(Function));
+    const [ cb ] = onAuthStateChangedSpy.calls.mostRecent().args;
+    cb(null);
```

Comment:
The more common case is the user is not null, but the `photoURL` is null, right? I think we should have test case for both.